File: fixtures/scale/php/small/src/Style/Theme.php
<?php

namespace TextPrint\Style;

require_once __DIR__ . '/AnsiPalette.php';

class Theme
{
    public string $name;
    public string $prefix;

    public static function named(string $name): self
    {
        $theme = new self();
        $theme->name = $name;
        $theme->prefix = match ($name) {
            'bright' => AnsiPalette::BOLD,
            'mono' => AnsiPalette::DIM,
            default => '',
        };
        return $theme;
    }

    public function apply(string $body): string
    {
        return $this->prefix === '' ? $body : $this->prefix . $body . AnsiPalette::RESET;
    }
}

File: fixtures/scale/csharp/small/Style/Theme.cs
using TextPrint.Style;

namespace TextPrint.Style
{
    public class Theme
    {
        public string Name { get; private set; }
        public string Prefix { get; private set; }

        public static Theme Named(string name)
        {
            var theme = new Theme { Name = name, Prefix = "" };
            if (name == "bright")
            {
                theme.Prefix = AnsiPalette.Bold;
            }
            else if (name == "mono")
            {
                theme.Prefix = AnsiPalette.Dim;
            }
            return theme;
        }

        public string Apply(string body)
        {
            if (Prefix.Length == 0)
            {
                return body;
            }
            return Prefix + body + AnsiPalette.Reset;
        }
    }
}
